cache.h

```diff
@@ -53,10 +61,11 @@ class Heightfield {
 class Waterfield : public Heightfield {
 public:
     float getHeight() const {
-      return waterHeight;
+      return liquidHeight;

```

Comment:
I know this is a small thing, but these kind of incremental renamings as features change are really important for the sake of readability/maintainability.

Thanks for paying attention to the little things, as well as the overall features.